CraftTweaker2-MC1120-Main/src/main/java/crafttweaker/mc1120/recipes/MCRecipeManager.java

```diff
@@ -301,132 +304,160 @@ public void removeRecipes(List<ResourceLocation> removingRecipes) {
             removingRecipes.forEach(recipe -> RegistryManager.ACTIVE.getRegistry(GameData.RECIPES).remove(recipe));
         }
     }
-    
-    public static class ActionReplaceAllOccurences extends ActionBaseRemoveRecipes {
-    
-        //I'm odd, in that I'm an ActionBaseRemoveRecipes, that also creates recipes.
+
+    public static class SubActionReplaceAllOccurences implements IAction {
+        private SubActionReplaceAllOccurences next;
         private final IIngredient toReplace;
         private final IIngredient replaceWith;
         private final IIngredient forOutput;
-        private List<MCRecipeBase> toChange;
-        private List<ResourceLocation> toRemove;
-        
-        public ActionReplaceAllOccurences(IIngredient toReplace, IIngredient replaceWith, IIngredient forOutput) {
+
+        public SubActionReplaceAllOccurences(IIngredient toReplace, IIngredient replaceWith, IIngredient forOutput) {
             this.toReplace = toReplace;
             this.replaceWith = replaceWith;
-            this.forOutput = forOutput == null ? IngredientAny.INSTANCE : forOutput;
+            this.forOutput = forOutput;
         }
-        
+
         @Override
         public void apply() {
-            toChange = getAllForIngredient(toReplace);
-            toRemove = toChange.stream()
-                    .map(f -> new ResourceLocation(f.getFullResourceName()))
-                    .collect(Collectors.toList());
-            removeRecipes(toRemove);
-            changeIngredients(toChange);
-        }
-        
-        @Override
-        public void removeRecipes(List<ResourceLocation> removingRecipes) {
-            List<ActionBaseAddRecipe> toUnAdd = new ArrayList<>();
-            removingRecipes.forEach(recipe -> {
-                RegistryManager.ACTIVE.getRegistry(GameData.RECIPES).remove(recipe);
-                recipesToAdd.stream()
-                        .filter(f -> f instanceof ActionDummyAddRecipe)
-                        .filter(f -> f.recipe.getRegistryName().equals(recipe))
-                        .forEach(toUnAdd::add);
-            });
-            toUnAdd.forEach(f -> {
-                recipesToAdd.remove(f);
-                usedRecipeNames.remove(f.getName());
-            });
+            if (ActionReplaceAllOccurences.INSTANCE.currentModifiedRecipe != null) {
```

Comment:
Please use early returns instead of nesting 5 times.
By that I mean something like

```java
if(ActionReplaceAllOccurences.INSTANCE.currentModifiedRecipe == null) {
    return;
}
```

If you invert the if-clause like that and use a return, then you don't need an "else" branch and everything is one nesting level higher, which makes it easier to read.